app/controllers/chat_channels_controller.rb

```diff
@@ -30,8 +30,12 @@ def create
   end
 
   def update
-    ChatChannelUpdateService.new(@chat_channel, chat_channel_params).update
-    flash[:settings_notice] = "Channel Settings Updated."
+    if ChatChannelUpdateService.new(@chat_channel, chat_channel_params).update
+      flash[:settings_notice] = "Channel Settings Updated."
+    else
+      flash[:settings_notice] = @chat_channel.valid? ? @chat_channel.errors.full_messages.to_sentence :
```

Comment:
Btw, you can set `flash[:error]` instead of `flash[:settings_notice]` here to be more clear that an error occured.

Reply:
To avoid checking for validity again you can set flash notice to `@chat_channel.errors.full_messages.to_sentence.presence || "error message"`

Reply:
updated